modules/gapi/test/common/gapi_core_tests_inl.hpp

```diff
@@ -1267,6 +1267,193 @@ TEST_P(NormalizeTest, Test)
     }
 }
 
+TEST_P(BackendOutputTest, EmptyOutput)
+{
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::GMat in1, in2, out;
+    out = cv::gapi::mul(in1, in2);
+    cv::GComputation c(cv::GIn(in1, in2), cv::GOut(out));
+    c.apply(cv::gin(in_mat1, in_mat2), cv::gout(out_mat_gapi), getCompileArgs());
+
+    // OpenCV code /////////////////////////////////////////////////////////////
+    cv::multiply(in_mat1, in_mat2, out_mat_ocv);
+
+    // Comparison //////////////////////////////////////////////////////////////
+    // Expected: output is allocated to the needed size
+    EXPECT_EQ(0, cv::countNonZero(out_mat_gapi != out_mat_ocv));
+    EXPECT_EQ(sz, out_mat_gapi.size());
+}
+
+TEST_P(BackendOutputTest, CorrectOutput)
```

Comment:
CorrectOutput -> TestOutputProperlyPreallocated ?

Reply:
changed name (to a bit different, though)